sdk/ai/ai-inference-rest/src/trace.ts

```diff
@@ -0,0 +1,101 @@
+    Request_Stop_Sequences = "gen_ai.request.stop_sequences",
+    Request_Temperature = "gen_ai.request.temperature",
+    Request_Top_K = "gen_ai.request.top_k",
+    Request_Top_P = "gen_ai.request.top_p",
+    Response_Finish_Reasons = "gen_ai.response.finish_reasons",
+    Response_Id = "gen_ai.response.id",
+    Response_Model = "gen_ai.response.model",
+    Usage_Input_Tokens = "gen_ai.usage.input_tokens",
+    Usage_Output_Tokens = "gen_ai.usage.output_tokens",
+    Server_Address = "server.address"
+  }
+
+  const getOperationName = (path: string) => {
+    switch (path) {
+      case "/chat/completions":
+        return "chat";
+      case "/info":
+        return "info";
+      case "/embeddings":
+        return "text_embeddings";
+      case "/images/embeddings":
+        return "image_embeddings";
+      default:
+        throw new Error(`Unknown path for span name: ${path}`);
+    }
+  }
+
+  const requestAttributeMapper = (request: RequestParameterWithBodyType) => {
+    const map = new Map<string, unknown>();
+    const body = (request as RequestParameterWithBodyType).body;
+
+    map.set(TracingAttributesEnum.Operation_Name, getOperationName(path));
+    map.set(TracingAttributesEnum.System, "az.ai_inference");
+    map.set(TracingAttributesEnum.Request_Model, body?.model);
+    map.set(TracingAttributesEnum.Request_Frequency_Penalty, body?.frequency_penalty);
+    map.set(TracingAttributesEnum.Request_Max_Tokens, body?.max_tokens);
+    map.set(TracingAttributesEnum.Request_Presence_Penalty, body?.presence_penalty);
+    map.set(TracingAttributesEnum.Request_Stop_Sequences, body?.stop);
+    map.set(TracingAttributesEnum.Request_Temperature, body?.temperature);
+    map.set(TracingAttributesEnum.Request_Top_P, body?.top_p);
+    return map;
+  }
+
+  const responseAttributeMapper = (_: RequestParameters, response?: PathUncheckedResponse, error?: unknown) => {
+    const map = new Map<string, unknown>();
+    if (error) {
+      if (error instanceof Error) {
+        map.set(TracingAttributesEnum.Error_Type, error.message);
+      } else {
+        map.set(TracingAttributesEnum.Error_Type, error);
+      }
+    }
+    if (response) {
+      let body = response.body;
+      map.set(TracingAttributesEnum.Response_Id, body.id);
+      map.set(TracingAttributesEnum.Response_Model, body.model);
+      if (body.usage) {
+        map.set(TracingAttributesEnum.Usage_Input_Tokens, body.usage.prompt_tokens);
+        map.set(TracingAttributesEnum.Usage_Output_Tokens, body.usage.completion_tokens);
+      }
+      if (body.error) {
+        map.set(TracingAttributesEnum.Error_Type, body.error.code);
+      }
+    }
+    return map;
+  }
+
+  const request = args as RequestParameterWithBodyType;
+
+  const name = `${getOperationName(path)} ${request.body?.model ?? ""}`;
+  return traceCLient.traceAsync(name, request, methodToTrace, requestAttributeMapper, responseAttributeMapper, options);
```

Comment:
not sure if it applies to sync and/or async API, but in the telemetry you've sent, the HTTP span or a sibling of `chat` span, but it should be a child.

I.e. you should have

```
main
   chat 
      mistral-....
        ...
```

and currently on your screenshot it's 

```
main
   chat 
   mistral-....
     ...
```

I believe the `traceCLient.traceAsync` should start the span as current, but it doesn't.

maybe @maorleger  can pinpoint the problem?

Reply:
@lmolkova   sorry..   perhaps I misunderstood the requirements.   Let me recap our conversation.
<img width="860" alt="image" src="https://github.com/user-attachments/assets/c6f2dde5-6775-4431-92fb-903294ad67be">
From my understanding, the chat and mistral don't need to be parent-child.   So is it a requirement?    I am afraid it is not a good time to investigate since http span creation has defects.   I have to comment out some code related to updateOptions in order to create http span.    With that said, something is incorrect with updateOptions, but creating chld span needs updateOptions.

Reply:
HTTP is expected to be a child of logical operation - this is the essential part of the tracing and also part of our policy https://azure.github.io/azure-sdk/general_implementation.html#general-tracing-new-span-per-rest-call

Reply:
you can't guarantee it will be, but you should make the best effort to

Reply:
Done